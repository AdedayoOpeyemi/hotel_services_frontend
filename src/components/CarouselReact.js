import React from 'react';
import { useSelector } from 'react-redux';
import Carousel from 'react-elastic-carousel';
import { Link } from 'react-router-dom';

function CarouselReact() {
    const services = useSelector((state) => state.services);
    const breakPoints = [
        { width: 1, itemsToShow: 1 },
        { width: 550, itemsToShow: 2, itemsToScroll: 2 },
        { width: 768, itemsToShow: 3 },
        { width: 1200, itemsToShow: 3 }
      ];

    return  (
        <div className="carouselcont">
          <div className="carousel-wrapper">
            <Carousel breakPoints={breakPoints}>
                    {services.services
        && services.services.map(({
          id, name, description, imageUrl,
        }, index) => (
          <div
            key={id} className="w-100 mx-3"
          >
            <Link to={`/services/${id}`}>
              <div className="card border-0">
                <img src={imageUrl} className="img-fluid imgcont" alt="..." />
                <div className="card-body text-center">
                  <h5 className="card-title text-center text-dark text-uppercase fw-bolder">{name}</h5>
                  <p className="card-text text-dark text-muted">{description}</p>
                  <p className="text-center fw-lighter text-muted">***************</p>
                </div>
              </div>
            </Link>
            <div className="text-center fs-3 row text-muted">
            <div className="col">
              <i className="fab fa-facebook" />
              <i className="fab fa-twitter px-3" />
              <i className="fab fa-instagram" />
            </div>
          </div>
          </div>
        ))}
            </Carousel>
          </div>
        </div>
      );
  }



export default CarouselReact;