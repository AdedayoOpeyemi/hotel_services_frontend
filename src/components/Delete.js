import React from 'react';

function Delete() {
  return (
    <h1 className="text-center">
      Delete services
    </h1>
  );
}

export default Delete;
