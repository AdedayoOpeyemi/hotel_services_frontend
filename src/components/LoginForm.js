import React from 'react';

const LoginForm = () => {
  const inputTextField = React.createRef();

  const printInput = (name) => {
    console.log(name);
  };

  return (
    <div className="mb-3">
      <input type="text" className="form-control" id="userName" aria-describedby="emailHelp" placeholder="User Name" ref={inputTextField} />
      <button className="btn btn-primary" type="submit" onClick={() => printInput(inputTextField.current.value)}>Button</button>
    </div>
  );
};

export default LoginForm;
