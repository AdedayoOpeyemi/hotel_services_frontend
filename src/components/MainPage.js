import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { useNavigate } from 'react-router-dom';
import { useSelector } from 'react-redux';
import Sidebar from './Sidebar';

const navbarOptions = {
  content: {
    bar: true,
    button: false,
  },
  form: {
    bar: false,
    button: true,
  },
};

const MainPage = ({ type, children }) => {
  const [showSidebar, setShowSidebar] = useState(navbarOptions[type]);
  const currentUser = useSelector((state) => state.user.user.userId);
  const navigate = useNavigate();

  if (currentUser === -1) navigate('/login');

  useEffect(() => {
    setShowSidebar(navbarOptions[type]);
  }, [type]);

  const toggleSidebar = () => {
    setShowSidebar(({ bar }) => ({
      bar: !bar,
      button: true,
    }));
  };

  return (
    <div className="d-flex">
      <button
        type="button"
        className={`btn green menubtn ${showSidebar.button ? '' : 'd-sm-none'}`}
        onClick={toggleSidebar}
      >
        <i className="fas fa-bars" />
      </button>
      <Sidebar show={showSidebar.bar} />
      <main className="container-fluid p-0">{children}</main>
    </div>
  );
};

MainPage.propTypes = {
  type: PropTypes.oneOf(['content', 'form']).isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.element,
  ]).isRequired,
};

export default MainPage;
