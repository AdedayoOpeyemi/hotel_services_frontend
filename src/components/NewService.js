import React from 'react';
import videos from './assets/office.mp4';

function NewService() {
  return (
    <header>
      <video autoPlay="autoplay" loop="loop" muted>
        <source src={videos} type="video/mp4" />
      </video>
      <div className="container text-white px-5">
        <div className="row vh-100 justify-content-center align-items-center">
          <div className="row">
            <h1 className="text-center mb-5">Create a new Service</h1>
            <div className="row">
              <div className="input-group mb-3">
                <div className="input-group-prepend" />
                <input type="text" className="form-control text-center rounded-pill mb-3" placeholder="Service name" aria-label="service_name" aria-describedby="basic-addon1" />
              </div>
            </div>
            <div className="row">
              <div className="input-group mb-3">
                <div className="input-group-prepend" />
                <input type="text" className="form-control text-center rounded-pill mb-3" placeholder="Service Description" aria-label="Service_description" aria-describedby="basic-addon1" />
              </div>
            </div>
            <div className="row">
              <div className="input-group mb-3">
                <div className="input-group-prepend" />
                <input type="number" className="form-control text-center rounded-pill mb-3" placeholder="Service Price" aria-label="Service_price" aria-describedby="basic-addon1" />
              </div>
            </div>
            <div className="row">
              <div className="input-group mb-3">
                <div className="input-group-prepend" />
                <input type="text" className="form-control text-center rounded-pill mb-3" placeholder="Service image URL" aria-label="Service_image" aria-describedby="basic-addon1" />
              </div>
            </div>
            <div className="row">
              <div className="col">
                <button type="button" className="btn text-white w-100 green fw-bolder rounded-pill mb-3">Create</button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </header>
  );
}

export default NewService;
