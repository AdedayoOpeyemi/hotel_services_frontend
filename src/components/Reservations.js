import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getReservationData, cancelReservationToApi } from '../redux/reservation/reservation_duck';

function Reservations() {
  const dispatch = useDispatch();
  const currentUser = useSelector((state) => state.user.user.userId);

  useEffect(() => {
    dispatch(getReservationData(currentUser));
  }, []);

  const reservations = useSelector((state) => state.reservations);

  return (
    <div className="pt-5">
      <h1 className="text-center mb-3">My Reservations</h1>
      { reservations
        && reservations.map(({
          reservation_id: id, date: { day, month, year },
          service_name: serviceName,
          service_description: serviceDescription, image_url: imageUrl, city,
        }) => (
          <div className="container" key={id}>
            <div className="card mb-3 border-0 bg-light">
              <div className="row g-0 ">
                <div className="col-md-4">
                  <img src={imageUrl} className="img-fluid rounded-start w-100" alt="..." />
                </div>
                <div className="col-md-8">
                  <div className="card-body text-center mt-lg-5">
                    <h2 className="card-title mb-3">{serviceName}</h2>
                    <ul className="list-group border-0 mb-5">
                      <li className="list-group-item list-group-item-dark border-0 mt-lg-5">{serviceDescription}</li>
                      <li className="list-group-item list-group-item-light border-0 listbg">
                        {day}
                        /
                        {month}
                        /
                        {year}
                      </li>
                      <li className="list-group-item list-group-item-dark border-0">{city}</li>
                    </ul>
                    <button
                      type="button"
                      className="btn w-50 green text-white fw-bolder rounded-pill mt-lg-5"
                      id={id}
                      onClick={() => dispatch(cancelReservationToApi(id, currentUser))}
                    >
                      Cancel Reservation
                    </button>
                  </div>
                </div>
              </div>
            </div>

          </div>
        ))}
      {' '}

    </div>
  );
}

export default Reservations;
