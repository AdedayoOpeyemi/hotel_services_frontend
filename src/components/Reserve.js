import React from 'react';
import videos from './assets/resort.mp4';

function Reserve() {
  return (
    <header>
      <video autoPlay="autoplay" loop="loop" muted>
        <source src={videos} type="video/mp4" />
      </video>
      <div className="container text-white">
        <div className="row vh-100 justify-content-center align-items-center">
          <div className="row">
            <h1 className="text-center">New Reservation</h1>
            <div className="row my-5">
              <div className="col">
                <div className="input-group mb-3">
                  <input type="text" className="form-control rounded-pill border-0" placeholder="Username" aria-label="Username" aria-describedby="basic-addon1" />
                </div>
              </div>
              <div className="col">
                <input type="text" className="form-control rounded-pill border-0" placeholder="Service Name" aria-label="service_name" aria-describedby="basic-addon1" />
              </div>
            </div>
            <div className="row">
              <div className="col">
                <div className="input-group mb-3 ">
                  <select className="form-select rounded-pill btng text-white border-0" id="year">
                    <option value="2021">2021</option>
                    <option value="2022">2022</option>
                    <option value="2023">2023</option>
                    <option value="2021">2024</option>
                    <option value="2022">2025</option>
                    <option value="2023">2026</option>
                  </select>
                </div>
              </div>
              <div className="col ">
                <div className="input-group mb-3">
                  <select className="form-select rounded-pill btng text-white border-0" id="month">
                    <option value="01">January</option>
                    <option value="02">February</option>
                    <option value="03">March</option>
                    <option value="04">April</option>
                    <option value="05">May</option>
                    <option value="06">June</option>
                    <option value="01">July</option>
                    <option value="02">August</option>
                    <option value="03">September</option>
                    <option value="04">October</option>
                    <option value="05">November</option>
                    <option value="06">December</option>
                  </select>
                </div>
              </div>
              <div className="col">
                <div className="input-group mb-3">
                  <select className="form-select rounded-pill btng text-white border-0" id="day">
                    <option value="01">01</option>
                    <option value="02">02</option>
                    <option value="03">03</option>
                    <option value="04">04</option>
                    <option value="05">05</option>
                    <option value="06">06</option>
                    <option value="01">07</option>
                    <option value="02">08</option>
                    <option value="03">09</option>
                    <option value="04">10</option>
                    <option value="05">11</option>
                    <option value="06">12</option>
                    <option value="01">13</option>
                    <option value="02">14</option>
                    <option value="03">15</option>
                    <option value="04">16</option>
                    <option value="05">17</option>
                    <option value="06">18</option>
                    <option value="01">19</option>
                    <option value="02">20</option>
                    <option value="03">21</option>
                    <option value="04">22</option>
                    <option value="05">23</option>
                    <option value="06">24</option>
                    <option value="02">25</option>
                    <option value="03">26</option>
                    <option value="04">27</option>
                    <option value="05">28</option>
                    <option value="06">29</option>
                    <option value="05">30</option>
                    <option value="06">31</option>
                  </select>
                </div>
              </div>
              <div className="col">
                <div className="input-group mb-3 ">
                  <select className="form-select rounded-pill btng text-white border-0" id="city">
                    <option value="London">London</option>
                    <option value="Melbourne">Melbourne</option>
                    <option value="Zurich">Zurich</option>
                    <option value="Tokyo">Tokyo</option>
                    <option value="Auckland">Auckland</option>
                    <option value="New York">New York</option>
                  </select>
                </div>
              </div>
              <div className="col">
                <button type="button" className="border-0 btn btn-light w-100 textg fw-bolder rounded-pill">Reserve</button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </header>
  );
}

export default Reserve;
