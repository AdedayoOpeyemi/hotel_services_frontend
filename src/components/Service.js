import React from 'react';
import { useNavigate, useParams } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { BiLeftArrow } from 'react-icons/bi';
import { currentService } from '../redux/service/service_duck';

const Service = () => {
  const { id } = useParams();
  const services = useSelector((state) => state.services);
  const service = services.services.find((service) => service.id.toString() === id);
  const navigate = useNavigate();
  const dispatch = useDispatch();

  if (service === undefined) navigate('/services');

  const toReserve = () => {
    dispatch(currentService(parseInt(id, 10)));
    navigate('/reserve');
  };

  return (
    <div className="container-fluid">
      <div className="row vh-100 justify-content-center align-items-center">
        <div className="card mb-3 border-0">
          <div className="row g-0">
            <div className="col-md-8 d-flex flex-column">
              <img src={service.imageUrl} className="img-fluid w-50 mx-auto" alt="..." />
              <button type="button" className="btn btn-light green text-white fw-bolder go-back mt-auto" aria-label="switch" onClick={() => navigate('/services')}>
                <BiLeftArrow className="imported-icon text-white" />
              </button>
            </div>
            <div className="col-md-4">
              <div className="card-body px-5">
                <h2 className="card-title text-center mb-lg-5">{service.name}</h2>
                <ul className="list-group border-0 mb-5">
                  <li className="list-group-item list-group-item-light border-0 listbg">{service.description}</li>
                  <li className="list-group-item list-group-item-dark border-0">{service.price}</li>
                  <li className="list-group-item list-group-item-light border-0 listbg">Service Duration 24 hours</li>
                </ul>
              </div>
              <div className="row justify-content-center">
                <button type="button" className="btn btn-light w-50 green text-white fw-bolder rounded-pill mb-3" onClick={() => toReserve()}>
                  Reserve
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Service;
