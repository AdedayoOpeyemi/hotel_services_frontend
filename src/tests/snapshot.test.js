

import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import {
  MemoryRouter as Router,
} from 'react-router-dom';
import { createMemoryHistory } from 'history';
import store from '../redux/configurateStore';
import CarouselReact from '../components/CarouselReact';
import LoginForm from '../components/LoginForm';
import MainPage from '../components/MainPage';
import NewService from '../components/NewService';
import Reservations from '../components/Reservations';
import Reserve from '../components/Reserve';
import Service from '../components/Service';
import Services from '../components/Services';
import Sidebar from '../components/Sidebar';
import Signup from '../components/Signup';
import Delete from '../components/Delete';
import './LocalStorageMock';

describe('Render Carousel', () => {
  it('Renders the Carousel', () => {
    const history = createMemoryHistory();
    history.push('/services');
    const carousel = TestRenderer.create(
      <Provider store={store}>
        <Router history={history}>
          <CarouselReact />
        </Router>
      </Provider>,
    ).toJSON();
    expect(carousel).toMatchSnapshot();
  });
});

describe('Render Delete page', () => {
  it('Renders the Delete page', () => {
    const history = createMemoryHistory();
    history.push('/delete');
    const del = TestRenderer.create(
      <Provider store={store}>
        <Router history={history}>
          <Delete />
        </Router>
      </Provider>,
    ).toJSON();
    expect(del).toMatchSnapshot();
  });
});

describe('Render LoginForm page', () => {
  it('Renders the LoginForm page', () => {
    const history = createMemoryHistory();
    history.push('/login');
    const logform = TestRenderer.create(
      <Provider store={store}>
        <Router history={history}>
          <LoginForm />
        </Router>
      </Provider>,
    ).toJSON();
    expect(logform).toMatchSnapshot();
  });
});

// describe('Render MainPage page', () => {
//   it('Renders the MainPage page', () => {
//     const mainpage = TestRenderer.create(
//       <Provider store={store}>
//         <MainPage />
//       </Provider>,
//     ).toJSON();
//     expect(mainpage).toMatchSnapshot();
//   });
// });

describe('Render NewService page', () => {
  it('Renders the NewService page', () => {
    const history = createMemoryHistory();
    history.push('/newservice');
    const newservice = TestRenderer.create(
      <Provider store={store}>
        <Router history={history}>
          <NewService />
        </Router>
      </Provider>,
    ).toJSON();
    expect(newservice).toMatchSnapshot();
  });
});

describe('Render Reservations page', () => {
  it('Renders the Reservations page', () => {
    const history = createMemoryHistory();
    history.push('/reservations');
    const reservations = TestRenderer.create(
      <Provider store={store}>
        <Router history={history}>
          <Reservations />
        </Router>
      </Provider>,
    ).toJSON();
    expect(reservations).toMatchSnapshot();
  });
});

describe('Render Reserve page', () => {
  it('Renders the Reserve page', () => {
    const history = createMemoryHistory();
    history.push('/reserve');
    const reserve = TestRenderer.create(
      <Provider store={store}>
        <Router history={history}>
          <Reserve />
        </Router>
      </Provider>,
    ).toJSON();
    expect(reserve).toMatchSnapshot();
  });
});

describe('Render Service page', () => {
  it('Renders the Service page', () => {
    const history = createMemoryHistory();
    history.push('/services/1');
    const service = TestRenderer.create(
      <Provider store={store}>
        <Router history={history}>
          <Service />
        </Router>
      </Provider>,
    ).toJSON();
    expect(service).toMatchSnapshot();
  });
});

describe('Render Services page', () => {
  it('Renders the Services page', () => {
    const history = createMemoryHistory();
    history.push('/services/1');
    const services = TestRenderer.create(
      <Provider store={store}>
        <Router history={history}>
          <Services />
        </Router>
      </Provider>,
    ).toJSON();
    expect(services).toMatchSnapshot();
  });
});

describe('Render Signup page', () => {
  it('Renders the Signup page', () => {
    const history = createMemoryHistory();
    history.push('/signup');
    const signup = TestRenderer.create(
      <Provider store={store}>
        <Router history={history}>
          <Signup />
        </Router>
      </Provider>,
    ).toJSON();
    expect(signup).toMatchSnapshot();
  });
});
