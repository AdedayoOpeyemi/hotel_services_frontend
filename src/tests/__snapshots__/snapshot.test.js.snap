// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Carousel Renders the Carousel 1`] = `
<div
  className="carouselcont"
>
  <div
    className="carousel-wrapper"
  >
    <div
      className="sc-jRQBWg qoLPS rec rec-carousel-wrapper "
      style={Object {}}
    >
      <div
        className="sc-eCImPb jMNKZk rec rec-carousel"
        size={
          Object {
            "height": 0,
          }
        }
        style={
          Object {
            "height": 0,
          }
        }
      >
        <button
          className="sc-bdvvtL cNNJIt rec rec-arrow rec rec-arrow-left"
          disabled={true}
          onClick={[Function]}
          style={Object {}}
          tabIndex={0}
          type="button"
        >
          ❮
        </button>
        <div
          className="sc-dkPtRN fPDpuU rec rec-slider-container"
        >
          <div
            className="sc-hKwDye jZlXPY rec rec-slider"
            style={
              Object {
                "left": "0px",
                "right": "auto",
                "top": "auto",
                "transition": "all 500ms ease",
              }
            }
          >
            <div
              className="rec rec-swipable"
              onMouseDown={[Function]}
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            />
          </div>
        </div>
        <button
          className="sc-bdvvtL cNNJIt rec rec-arrow rec rec-arrow-right"
          disabled={true}
          onClick={[Function]}
          style={Object {}}
          tabIndex={0}
          type="button"
        >
          ❯
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Render Delete page Renders the Delete page 1`] = `
<div
  className="pt-5"
>
  <h1
    className="text-center mb-3"
  >
    Delete Service
  </h1>
   
</div>
`;

exports[`Render LoginForm page Renders the LoginForm page 1`] = `
<header>
  <video
    autoPlay="autoplay"
    loop="loop"
    muted={true}
  >
    <source
      src="pool.mp4"
      type="video/mp4"
    />
  </video>
  <div
    className="container"
  >
    <div
      className="row vh-100 justify-content-center align-items-center"
    >
      <div
        className="alert alert-warning alert-dismissible fade show d-none"
        role="alert"
      >
        <strong>
          Holy guacamole!
        </strong>
         
        You should check in on some of those fields below.
        <button
          aria-label="Close"
          className="btn-close"
          data-bs-dismiss="alert"
          type="button"
        />
      </div>
      <div
        className="col text-center text-white"
      >
        <h1>
          Welcome to City Services
        </h1>
        <h5
          className="fs-6 fw-normal"
        />
        <input
          aria-describedby="emailHelp"
          className="form-control my-3"
          id="userName"
          placeholder="User Name"
          type="text"
        />
        <button
          className="btn green mx-3"
          onClick={[Function]}
          type="button"
        >
          Login
        </button>
        <a
          href="/signup"
          onClick={[Function]}
        >
          <button
            className="btn green"
            type="button"
          >
            Sign up
          </button>
        </a>
      </div>
    </div>
  </div>
</header>
`;

exports[`Render MainPage content page Renders the MainPage page 1`] = `
<div
  className="d-flex"
>
  <button
    className="btn green menubtn d-sm-none"
    onClick={[Function]}
    type="button"
  >
    <i
      className="fas fa-bars"
    />
  </button>
  <div
    className="sidebarcont sidebarbg h-100 "
  >
    <nav
      className="nav-menu vh-100 pt-5 "
    >
      <div
        className="row h-100 ps-3 "
      >
        <h1
          className="sidebarfont rotate text-center"
        >
          Services
        </h1>
        <div
          className="list-group border-0 "
        >
          <a
            className="nav-link text-dark sidefont"
            href="/services"
            onClick={[Function]}
          >
            Services
          </a>
          <a
            className="nav-link text-dark sidefont"
            href="/reserve"
            onClick={[Function]}
          >
            Reserve
          </a>
          <a
            className="nav-link text-dark sidefont"
            href="/reservations"
            onClick={[Function]}
          >
            My Reservations
          </a>
          <a
            className="nav-link text-dark sidefont"
            href="/newservice"
            onClick={[Function]}
          >
            Add Services
          </a>
          <a
            className="nav-link text-dark sidefont"
            href="/delete"
            onClick={[Function]}
          >
            Delete service
          </a>
        </div>
        <div
          className="text-center row mt-auto pb-5"
        >
          <a
            className="nav-link text-dark sidefont"
            href="/login"
            onClick={[Function]}
          >
            Log out
          </a>
          <div
            className="col fs-3"
          >
            <i
              className="fab fa-facebook"
            />
            <i
              className="fab fa-twitter px-3"
            />
            <i
              className="fab fa-instagram"
            />
          </div>
        </div>
      </div>
    </nav>
  </div>
  <main
    className="container-fluid p-0"
  >
    <div />
  </main>
</div>
`;

exports[`Render MainPage form page Renders the MainPage page 1`] = `
<div
  className="d-flex"
>
  <button
    className="btn green menubtn "
    onClick={[Function]}
    type="button"
  >
    <i
      className="fas fa-bars"
    />
  </button>
  <div
    className="sidebarcont sidebarbg h-100 d-none"
  >
    <nav
      className="nav-menu vh-100 pt-5 "
    >
      <div
        className="row h-100 ps-3 "
      >
        <h1
          className="sidebarfont rotate text-center"
        >
          Services
        </h1>
        <div
          className="list-group border-0 "
        >
          <a
            className="nav-link text-dark sidefont"
            href="/services"
            onClick={[Function]}
          >
            Services
          </a>
          <a
            className="nav-link text-dark sidefont"
            href="/reserve"
            onClick={[Function]}
          >
            Reserve
          </a>
          <a
            className="nav-link text-dark sidefont"
            href="/reservations"
            onClick={[Function]}
          >
            My Reservations
          </a>
          <a
            className="nav-link text-dark sidefont"
            href="/newservice"
            onClick={[Function]}
          >
            Add Services
          </a>
          <a
            className="nav-link text-dark sidefont"
            href="/delete"
            onClick={[Function]}
          >
            Delete service
          </a>
        </div>
        <div
          className="text-center row mt-auto pb-5"
        >
          <a
            className="nav-link text-dark sidefont"
            href="/login"
            onClick={[Function]}
          >
            Log out
          </a>
          <div
            className="col fs-3"
          >
            <i
              className="fab fa-facebook"
            />
            <i
              className="fab fa-twitter px-3"
            />
            <i
              className="fab fa-instagram"
            />
          </div>
        </div>
      </div>
    </nav>
  </div>
  <main
    className="container-fluid p-0"
  >
    <div />
  </main>
</div>
`;

exports[`Render NewService page Renders the NewService page 1`] = `
<header>
  <img
    alt="..."
    src="new-service-background.jpg"
  />
  <div
    className="container text-white px-5"
  >
    <div
      className="row vh-100 justify-content-center align-items-center"
    >
      <div
        className="row"
      >
        <h1
          className="text-center mb-3"
        >
          Create a new Service
        </h1>
        <p
          className="text-center mb-4"
        >
          Add a great service to our already impressive catalog!
        </p>
        <div
          className="row"
        >
          <div
            className="mb-3"
          >
            <div
              className="input-group"
            >
              <div
                className="input-group-prepend"
              />
              <input
                aria-describedby="basic-addon1"
                aria-label="service_name"
                className="form-control text-center rounded-pill mb-3"
                onChange={[Function]}
                placeholder="Service name"
                type="text"
              />
            </div>
            
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="mb-3"
          >
            <div
              className="input-group"
            >
              <div
                className="input-group-prepend"
              />
              <input
                aria-describedby="basic-addon1"
                aria-label="Service_description"
                className="form-control text-center rounded-pill mb-3"
                onChange={[Function]}
                placeholder="Service Description"
                type="text"
              />
            </div>
            
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="mb-3"
          >
            <div
              className="input-group"
            >
              <div
                className="input-group-prepend"
              />
              <input
                aria-describedby="basic-addon1"
                aria-label="Service_price"
                className="form-control text-center rounded-pill mb-3"
                min="0"
                onChange={[Function]}
                placeholder="Service Price"
                type="number"
              />
            </div>
            
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="mb-3"
          >
            <div
              className="input-group"
            >
              <div
                className="input-group-prepend"
              />
              <input
                aria-describedby="basic-addon1"
                aria-label="Service_image"
                className="form-control text-center rounded-pill mb-3 d-none"
                onChange={[Function]}
                placeholder="Service image URL"
                type="text"
                value="https://picsum.photos/id/237/500"
              />
            </div>
            
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col"
          >
            <button
              className="btn btn-light w-100 textg fw-bolder rounded-pill mb-3"
              onClick={[Function]}
              type="button"
            >
              Create
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;

exports[`Render Reservations page Renders the Reservations page 1`] = `
<div
  className="pt-5"
>
  <h1
    className="text-center mb-3"
  >
    My Reservations
  </h1>
   
</div>
`;

exports[`Render Reserve page Renders the Reserve page 1`] = `
<header>
  <img
    alt="..."
    src="new-reservation-background.jpg"
  />
  <div
    className="container text-white"
  >
    <div
      className="row vh-100 justify-content-center align-items-center"
    >
      <div
        className="row"
      >
        <h1
          className="text-center"
        >
          New Reservation
        </h1>
        <p
          className="text-center mb-3"
        >
          Our hotels have best-in-class services. Pick an individual service for a luxurious stay or a social service to stand out.
        </p>
        <div
          className="row my-3"
        >
          <div
            className="col-md-6"
          >
            <div
              className="input-group mb-3"
            >
              <input
                aria-describedby="basic-addon1"
                aria-label="Username"
                className="form-control rounded-pill border-0 text-center"
                disabled={true}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="col-md-6"
          >
            <select
              className="form-select rounded-pill border-0"
              defaultValue={-1}
              id="service-name"
              onChange={[Function]}
            >
              <option
                disabled={true}
                value={-1}
              >
                Choose a service to reserve
              </option>
            </select>
            
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-md-2"
          >
            <div
              className="input-group mb-3 "
            >
              <select
                className="form-select rounded-pill btng text-white border-0"
                id="year"
                onChange={[Function]}
              >
                <option
                  value={2021}
                >
                  2021
                </option>
                <option
                  value={2022}
                >
                  2022
                </option>
                <option
                  value={2023}
                >
                  2023
                </option>
                <option
                  value={2024}
                >
                  2024
                </option>
                <option
                  value={2025}
                >
                  2025
                </option>
                <option
                  value={2026}
                >
                  2026
                </option>
              </select>
              
            </div>
          </div>
          <div
            className="col-md-3 "
          >
            <div
              className="input-group mb-3"
            >
              <select
                className="form-select rounded-pill btng text-white border-0"
                id="month"
                onChange={[Function]}
              >
                <option
                  value={12}
                >
                  December
                </option>
              </select>
            </div>
          </div>
          <div
            className="col-md-2"
          >
            <div
              className="input-group mb-3"
            >
              <select
                className="form-select rounded-pill btng text-white border-0"
                id="day"
                onChange={[Function]}
              >
                <option
                  value={2}
                >
                  2
                </option>
                <option
                  value={3}
                >
                  3
                </option>
                <option
                  value={4}
                >
                  4
                </option>
                <option
                  value={5}
                >
                  5
                </option>
                <option
                  value={6}
                >
                  6
                </option>
                <option
                  value={7}
                >
                  7
                </option>
                <option
                  value={8}
                >
                  8
                </option>
                <option
                  value={9}
                >
                  9
                </option>
                <option
                  value={10}
                >
                  10
                </option>
                <option
                  value={11}
                >
                  11
                </option>
                <option
                  value={12}
                >
                  12
                </option>
                <option
                  value={13}
                >
                  13
                </option>
                <option
                  value={14}
                >
                  14
                </option>
                <option
                  value={15}
                >
                  15
                </option>
                <option
                  value={16}
                >
                  16
                </option>
                <option
                  value={17}
                >
                  17
                </option>
                <option
                  value={18}
                >
                  18
                </option>
                <option
                  value={19}
                >
                  19
                </option>
                <option
                  value={20}
                >
                  20
                </option>
                <option
                  value={21}
                >
                  21
                </option>
                <option
                  value={22}
                >
                  22
                </option>
                <option
                  value={23}
                >
                  23
                </option>
                <option
                  value={24}
                >
                  24
                </option>
                <option
                  value={25}
                >
                  25
                </option>
                <option
                  value={26}
                >
                  26
                </option>
                <option
                  value={27}
                >
                  27
                </option>
                <option
                  value={28}
                >
                  28
                </option>
                <option
                  value={29}
                >
                  29
                </option>
                <option
                  value={30}
                >
                  30
                </option>
                <option
                  value={31}
                >
                  31
                </option>
              </select>
            </div>
          </div>
          <div
            className="col-md-3"
          >
            <div
              className="input-group mb-3 "
            >
              <select
                className="form-select rounded-pill btng text-white border-0"
                defaultValue=""
                id="city"
                onChange={[Function]}
              >
                <option
                  disabled={true}
                  value=""
                >
                  Please, select a city
                </option>
                <option
                  value="London"
                >
                  London
                </option>
                <option
                  value="Melbourne"
                >
                  Melbourne
                </option>
                <option
                  value="Zurich"
                >
                  Zurich
                </option>
                <option
                  value="Tokyo"
                >
                  Tokyo
                </option>
                <option
                  value="Auckland"
                >
                  Auckland
                </option>
                <option
                  value="New York"
                >
                  New York
                </option>
              </select>
              
            </div>
          </div>
          <div
            className="col-md-2"
          >
            <button
              className="border-0 btn btn-light w-100 textg fw-bolder rounded-pill"
              onClick={[Function]}
              type="button"
            >
              Reserve
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;

exports[`Render Services page Renders the Services page 1`] = `
<div
  className="container-fluid vh-100 d-flex justify-content-center align-items-center"
>
  <div
    className="row w-100"
  >
    <h1
      className="text-center text-uppercase fw-bolder tittle"
    >
      All Services
    </h1>
    <p
      className="fw-lighter text-center mb-3 fw-bold grey"
    >
      Please select a Service
    </p>
    <p
      className="text-center fw-lighter grey"
    >
      ***************
    </p>
    <div
      className="carouselcont"
    >
      <div
        className="carousel-wrapper"
      >
        <div
          className="sc-jRQBWg qoLPS rec rec-carousel-wrapper "
          style={Object {}}
        >
          <div
            className="sc-eCImPb jMNKZk rec rec-carousel"
            size={
              Object {
                "height": 0,
              }
            }
            style={
              Object {
                "height": 0,
              }
            }
          >
            <button
              className="sc-bdvvtL cNNJIt rec rec-arrow rec rec-arrow-left"
              disabled={true}
              onClick={[Function]}
              style={Object {}}
              tabIndex={0}
              type="button"
            >
              ❮
            </button>
            <div
              className="sc-dkPtRN fPDpuU rec rec-slider-container"
            >
              <div
                className="sc-hKwDye jZlXPY rec rec-slider"
                style={
                  Object {
                    "left": "0px",
                    "right": "auto",
                    "top": "auto",
                    "transition": "all 500ms ease",
                  }
                }
              >
                <div
                  className="rec rec-swipable"
                  onMouseDown={[Function]}
                  style={
                    Object {
                      "display": "flex",
                      "flexDirection": "row",
                    }
                  }
                />
              </div>
            </div>
            <button
              className="sc-bdvvtL cNNJIt rec rec-arrow rec rec-arrow-right"
              disabled={true}
              onClick={[Function]}
              style={Object {}}
              tabIndex={0}
              type="button"
            >
              ❯
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Render Signup page Renders the Signup page 1`] = `
<header>
  <video
    autoPlay="autoplay"
    loop="loop"
    muted={true}
  >
    <source
      src="pool.mp4"
      type="video/mp4"
    />
  </video>
  <div
    className="container"
  >
    <div
      className="row vh-100 justify-content-center align-items-center"
    >
      <div
        className="alert alert-warning alert-dismissible fade show d-nonce d-none"
        role="alert"
      >
        <strong>
          Holy guacamole!
        </strong>
         
        You should check in on some of those fields below.
        <button
          aria-label="Close"
          className="btn-close"
          data-bs-dismiss="alert"
          type="button"
        />
      </div>
      <div
        className="col text-center text-white"
      >
        <h1>
          Welcome to City Services
        </h1>
        <h5
          className="fs-6 fw-normal"
        >
          Please Sign up using your username
        </h5>
        <input
          aria-describedby="emailHelp"
          className="form-control my-3"
          id="userName"
          placeholder="User Name"
          type="text"
        />
        <button
          className="btn green"
          onClick={[Function]}
          type="button"
        >
          Create User
        </button>
      </div>
    </div>
  </div>
</header>
`;
